Set/Models/GameModel.swift

```diff
@@ -20,23 +40,25 @@ struct GameModel {
         playedCards = []
     }
     
-    mutating func dealThreeMoreCards() {
-        guard !deck.isEmpty else { return }
-        
-        if deck.count >= 3 {
-            for _ in 1...3 {
-                let removedCard = deck.removeFirst()
-                cardsOnTheScreen.append(removedCard)
-            }
-        } else if deck.count < 3 {
-            for _ in 1...deck.count {
-                let removedCard = deck.removeFirst()
-                cardsOnTheScreen.append(removedCard)
-            }
+    private mutating func resetGame() {
+        deck = []
+        cardsOnTheScreen = []
+        playerCards = []
+        playedCards = []
+    }
+    
+    mutating func checkASet(setOfCards: [CardModel]) {
+        print("checkASet func is called")
+        for card in setOfCards {
+            print("checked")
         }
     }
     
-//    func createNewGame() -> GameModel {
-//
-//    }
+    
+    mutating func toggle(cardId: UUID) {
+        guard let chousenIndex = cardsOnTheScreen.firstIndex(where: {$0.id == cardId}) else {
```

Comment:
It seems that you store `isActive` or selected state in two places, either you should do it in `GameModel` and everytime filter card where `isActive` is true, or card itself doesn't know about it's state in cardCade you add this state on the level of ViewModel.